src/freertos_drivers/esp32/Esp32WiFiManager.cxx

```diff
@@ -372,93 +407,238 @@ void Esp32WiFiManager::factory_reset(int fd)
+            // Clear the flag that indicates we are connected to the SSID.
+            xEventGroupClearBits(wifiStatusEventGroup_, WIFI_CONNECTED_BIT);
+            // Clear the flag that indicates we have an IPv4 address.
+            xEventGroupClearBits(wifiStatusEventGroup_, WIFI_GOTIP_BIT);
+
+            // Wake up the wifi_manager_task so it can clean up
+            // connections.
+            xTaskNotifyGive(wifiTaskHandle_);
+        }
+
+        // If we are managing the WiFi and MDNS systems we need to
+        // trigger the reconnection process at this point.
+        if (manageWiFi_)
+        {
+            if (was_previously_connected)
             {
                 LOG(INFO, "[WiFi] Attempting to reconnect to SSID: %s.",
                     ssid_);
-                esp_wifi_connect();
             }
-            break;
+            else
+            {
+                LOG(INFO,
+                    "[WiFi] Connection failed, reconnecting to SSID: %s.",
+                    ssid_);
+            }
+            esp_wifi_connect();
+        }
+    }
+    else if (event->event_id == SYSTEM_EVENT_AP_START && manageWiFi_)
+    {
+        // Set the generated hostname prior to connecting to the SSID
+        // so that it shows up with the generated hostname instead of
+        // the default "Espressif".
+        LOG(INFO, "[SoftAP] Setting ESP32 hostname to \"%s\".",
+            hostname_.c_str());
+        ESP_ERROR_CHECK(tcpip_adapter_set_hostname(
+            TCPIP_ADAPTER_IF_AP, hostname_.c_str()));
+
+        uint8_t mac[6];
+        esp_wifi_get_mac(WIFI_IF_AP, mac);
+        LOG(INFO, "[SoftAP] MAC Address: %s", mac_to_string(mac).c_str());
+
+        if (softAPStaticIP_ && wifiMode_ != WIFI_MODE_STA)
+        {
+            // Stop the DHCP server so we can reconfigure it.
+            LOG(INFO, "[SoftAP] Stoping DHCP Server (if running).");
+            ESP_ERROR_CHECK(tcpip_adapter_dhcps_stop(TCPIP_ADAPTER_IF_AP));
+
+            LOG(INFO,
+                "[SoftAP] Configuring Static IP address:\n"
+                "IP     : " IPSTR "\n"
+                "Gateway: " IPSTR "\n"
+                "Netmask: " IPSTR,
+                IP2STR(&softAPStaticIP_->ip),
+                IP2STR(&softAPStaticIP_->gw),
+                IP2STR(&softAPStaticIP_->netmask));
+            ESP_ERROR_CHECK(
+                tcpip_adapter_set_ip_info(TCPIP_ADAPTER_IF_AP,
+                                          softAPStaticIP_));
+
+            // Convert the Soft AP Static IP to a uint32 for manipulation
+            uint32_t apIP = ntohl(ip4_addr_get_u32(&softAPStaticIP_->ip));
+
+            // Default configuration is for DHCP addresses to follow
+            // immediately after the static ip address of the Soft AP.
+            ip4_addr_t first_ip, last_ip;
+            ip4_addr_set_u32(&first_ip, htonl(apIP + 1));
+            ip4_addr_set_u32(&last_ip, htonl(apIP + softAPMaxStations_));
```

Comment:
reserve a few dozen IPs instead. If someone connects a wifi extender you get to deal with a lot more MAC addresses than 4.

Reply:
Esp32 only supports a max of 4 connected stations. I can increase it though.

Reply:
The number of stations does not necessarily equal to the number of MAC addresses visible. If you add a wifi extender, you will see only one station (the wifi extender) but a bunch of MAC addresses that need separate IP addresses.

Reply:
got it, setting up a constant with a default value of 48 and capping at max of 4 stations (max for esp32).